### PR TITLE
Use new Pool terminology

### PR DESCRIPTION
**Pull request by IainM22**
The thinking here is that for the v2 contracts we can create a new mapping file called `pool.ts` (or something else).  Any events that will have new event signatures like stake/unstake can have separate mappings, one in geyserv1 and one in the new `pool.ts`. Any mappings that can be shared between the two like handleGysrSpent can move from `geyserV1.ts` to a third ‘common’ mappings file.

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -64,7 +64,7 @@ type Geyser @entity {
 
   start: BigInt!
   end: BigInt!
-  state: GeyserState!
+  state: PoolState!
 
   stakedUSD: BigDecimal!
   rewardsUSD: BigDecimal!
```
Comment (devinaconley) [inline schema.graphql:70]: can we also rename `apy` to `apr` while we're breaking stuff?

File: src/pricing/pool.ts [hunk 2]
```diff
@@ -65,33 +65,33 @@ export function updatePricing(
...
 
-  geyser.sharesPerSecond = rate;
-  geyser.stakingSharesPerToken = stakingSharesPerToken;
-  geyser.rewardSharesPerToken = rewardSharesPerToken;
+  pool.sharesPerSecond = rate;
+  pool.stakingSharesPerToken = stakingSharesPerToken;
+  pool.rewardSharesPerToken = rewardSharesPerToken;
 
 
   // apy
```
Comment (devinaconley) [inline src/pricing/pool.ts]: apr

File: subgraph.template.yaml [hunk 3]
```diff
@@ -38,23 +38,23 @@ dataSources:
       file: ./src/mappings/platform.ts
 
   - kind: ethereum/contract
-    name: GeyserFactory
+    name: GeyserV1Factory
```
Comment (devinaconley) [inline subgraph.template.yaml]: Can we make this `GeyserFactoryV1`?

File: src/mappings/platform.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 // platform wide mappings
 
 import { Address, BigInt, log, ethereum } from '@graphprotocol/graph-ts'
-import { Geyser as GeyserContract } from '../../generated/templates/Geyser/Geyser'
-import { Geyser, Token, Platform } from '../../generated/schema'
+import { GeyserV1 as GeyserV1Contract } from '../../generated/templates/GeyserV1/GeyserV1'
```
Comment (devinaconley) [inline src/mappings/platform.ts]: Can we import this as `GeyserContractV1`?

File: src/mappings/geyserv1.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
-// Geyser event handling and mapping
+// Pool event handling and mapping
 
 import { BigInt, log, store } from '@graphprotocol/graph-ts'
 import {
-  Geyser as GeyserContract,
+  GeyserV1 as GeyserV1Contract,
```
Comment (devinaconley) [inline src/mappings/geyserv1.ts]: Can we import this as `GeyserContractV1`?

File: src/mappings/factory.ts [hunk 6]
```diff
@@ -1,20 +1,19 @@
...
 import { Address, BigInt, BigDecimal, log } from '@graphprotocol/graph-ts'
-import { GeyserFactory, GeyserCreated } from '../../generated/GeyserFactory/GeyserFactory'
-import { Geyser as GeyserContract } from '../../generated/GeyserFactory/Geyser'
-import { ERC20 } from '../../generated/GeyserFactory/ERC20'
-import { Geyser, Platform, Token, User } from '../../generated/schema'
-import { Geyser as GeyserTemplate } from '../../generated/templates'
+import { GeyserCreated } from '../../generated/GeyserV1Factory/GeyserV1Factory'
+import { GeyserV1 as GeyserV1Contract } from '../../generated/GeyserV1Factory/GeyserV1'
+import { Pool, Platform, Token, User } from '../../generated/schema'
+import { GeyserV1 as GeyserV1Template } from '../../generated/templates'
```
Comment (devinaconley) [inline src/mappings/factory.ts]: Can we import this as `GeyserTemplateV1`?

File: src/mappings/factory.ts [hunk 7]
```diff
@@ -1,20 +1,19 @@
 // GYSR factory event handling and mapping
```
Comment (devinaconley) [inline src/mappings/factory.ts]: should we rename this to `factoryv1.ts`?

File: src/pricing/pool.ts [hunk 8]
```diff
@@ -1,32 +1,32 @@
 // pricing for geyser information on apy, tvl, and more
```
Comment (devinaconley) [inline src/pricing/pool.ts]: *pool

**Review by devinaconley — CHANGES_REQUESTED** [review]
This is looking good :ok_hand:

Just some minor comments on style

File: src/mappings/factory.ts (hunk 7 above)
Comment (IainM22) [reply src/mappings/factory.ts]: Yeah I think thats a good idea

File: schema.graphql (hunk 1 above)
Comment (IainM22) [reply schema.graphql:70]: Probably an appropriate time to do that

**Review by devinaconley — APPROVED** [review]
looks great, thanks! :+1:

**Pull request merged**